portal-impl/src/com/liferay/portal/minifier/MinifierUtil.java

```diff
@@ -50,6 +62,32 @@ public static String minifyJavaScript(String resourceName, String content) {
 		return content;
 	}
 
+	private static String _getServletContextPath(String resourceName) {
+		String pathModule = Portal.PATH_MODULE;
+
+		int i = resourceName.indexOf(StringPool.SLASH, pathModule.length() + 1);
+
+		if (i == -1) {
+			return StringPool.BLANK;
+		}
+
+		return resourceName.substring(0, i);
+	}
+
+	private static boolean _isLiferayResource(String resourceName) {
+		if (resourceName.startsWith(
+				Portal.PATH_MODULE + "/js/resolved-module/") ||
+			resourceName.startsWith(Portal.PATH_MODULE + "/classic-theme/") ||
+			resourceName.startsWith(Portal.PATH_MODULE + "/admin-theme/") ||
+			_liferayServletContextsMap.containsKey(
+				_getServletContextPath(resourceName))) {
+
+			return true;
+		}
+
+		return false;
```

Comment:
:shrug: That's why I said in slack that this may be incomplete. AFAIK we cover everything, but some exotic things may still be left out and be minified when they shouldn't.

AFAIK everything is covered, because all DXP bundles have symbolic name starting with `com.liferay` and only themes, which are WAR files, are out of that rule. Then, the only place where we use virtual URLs is the AMD machinery (`js/resolved-module`) and combo (but combo delegates to `js/resolved-module` to minify, so it should be OK.

Reply:
There would be a second part for this feature that is related to minifying the content coming from JSPs, but since we cannot determine the source of those snippets because the minifying filter (`StripFilter`) only sees the full resulting HTML @bryceosterhaus and me decided to not implement it for now. Especially since we decided that all that JS code inside JSPs should be moved to .js files.